node/node.go

```diff
@@ -119,9 +120,10 @@ type Node struct {
 	clientServer *clientService.Server
 
 	// Syncing component.
-	downloaderServer *downloader.Server
-	stateSync        *syncing.StateSync
-	syncingState     uint32
+	downloaderServer       *downloader.Server
+	stateSync              *syncing.StateSync
+	syncingState           uint32
+	peerRegistrationRecord map[string]int64 // record registration time (unixtime) of peers begin in syncing
```

Comment:
You may have map concurrent access issue. please consider using sync.Map data structure.

Reply:
I will add mutex for map success